### PR TITLE
Preserve tags fields simple selector

### LINKED ISSUE #371: Preserve tags to join/window
Hi,

I am creating tick script with measurement with tags (server_group, dc, etc), my 
tick script is something like this:

```javascript
var windows = stream.from('some_measurement')
									  .where(lambda: 'dc' = 'europe')
										.window()
											.every(10s)
											.period(40s)

var first = windows.first('value')
var last = windows.last('value')


first.join(last)
		 .eval(lambda: 'last.last' - 'first.first').as('cvalue')
		 .alert()
			// some levels..
			.post('http://some-service')
```

In the json I am getting on the service I don't have all tags I have in "some_measurement" whom I need.
Is there a way to preserve the tags?

### PR DESCRIPTION
**Pull request by yosiat**
Fixes #371

### CODE REVIEW

**Comment by yosiat** [comment]
This pull request and #422 are failing because of:
```
Downloading vet tool...
[DEBUG] go get golang.org/x/tools/cmd/vet


Executed command failed!
-- Command run was: go get golang.org/x/tools/cmd/vet
-- Failure was: package golang.org/x/tools/cmd/vet: cannot find package "golang.org/x/tools/cmd/vet" in any of:
	/usr/local/go/src/golang.org/x/tools/cmd/vet (from $GOROOT)
	/root/go/src/golang.org/x/tools/cmd/vet (from $GOPATH)
```

If I am not mistaken the "go tool vet" is baked in go and there is no need to "go get" it.

File: influxql.gen.go.tmpl [hunk 1]
```diff
@@ -125,7 +143,7 @@ func (e *{{.name}}PointEmitter) EmitPoint() (models.Point, error) {
 		Time:       t,
 		Group:      e.group,
 		Dimensions: e.dimensions,
-		Tags:       e.tags,
+		Tags:       ap.Aux[0].(models.Tags),
```
Comment (nathanielc) [inline influxql.gen.go.tmpl]: This looks like the cause of the CI failure. ap.Aux[0] will only have the tags if isSimpleSelector is true.

File: influxql.gen.go.tmpl [hunk 2]
```diff
@@ -16,6 +16,7 @@ import (
 type {{.name}}PointAggregator struct {
 	field         string
 	topBottomInfo *pipeline.TopBottomCallInfo
+  isSimpleSelector bool
```
Comment (nathanielc) [inline influxql.gen.go.tmpl]: Can you use tabs in the `tmpl` files as if there were real go source? Thanks

**Comment by nathanielc** [comment]
@yosiat Other than those two small issues this looks good to go. Thanks!

**Comment by yosiat** [comment]
@nathanielc , I added a simple condition to fetch from the tags from the right place (aux/emitter's tags).
But I found out that Aux is empty for this test: TestStream_InfluxQL the Percentile test case specifically.

I see that in floatPointBulkAggregator.AggregateBatch, I do put the tags in the aux fields, but on floatPointEmitter.EmitPoint the Aux is empty.

Do you know why?

**Comment by nathanielc** [comment]
@yosiat  I took a look at the percentile case. Looks like a bug on the InfluxDB side https://github.com/influxdata/influxdb/blob/master/influxql/call_iterator.go#L991

The returned point does not preserve Aux fields or Tags. I'll ping them and see what they say. It should be an easy fix on their end of things.

**Comment by nathanielc** [comment]
This is the PR for fixing percentile in InfluxDB https://github.com/influxdata/influxdb/pull/6293

**Comment by yosiat** [comment]
@nathanielc I tried to run the tests after doing "git pull" in $GOPATH/github.com/influxdata/influxdb but I am getting those errors:
```
pipeline/influxql.go:89: undefined: "github.com/influxdata/influxdb/influxql".FloatDistinctReduceSlice
pipeline/influxql.go:93: undefined: "github.com/influxdata/influxdb/influxql".IntegerDistinctReduceSlice
```

I did something wrong, or should I fix those errors?

**Comment by nathanielc** [comment]
> I did something wrong, or should I fix those errors?

Nope, InfluxDB core changed again and broke things. I am working on setting up vendoring so that things like this don't keep breaking Kapacitor builds.

**Comment by nathanielc** [comment]
@yosiat Vendoring complete, a rebase should fix the build.

**Comment by yosiat** [comment]
@nathanielc for some reason I fail to run the tests:
```
λ ~/code/go/src/github.com/influxdata/kapacitor/ preserve-tags-fields-simple-selector go test -v -run='TestStream_InfluxQL' ./integrations/streamer_test.go ./integrations/helpers_test.go
# command-line-arguments
integrations/streamer_test.go:20:2: cannot find package "command-l/vendor/github.com/influxdata/influxdb/client" in any of:
	/Users/yosi/.gvm/gos/go1.6/src/command-l/vendor/github.com/influxdata/influxdb/client (from $GOROOT)
	/Users/yosi/code/go/src/command-l/vendor/github.com/influxdata/influxdb/client (from $GOPATH)
FAIL	command-line-arguments [setup failed]
```

do you have any idea why?

**Comment by nathanielc** [comment]
@yosiat Not sure, what is `command-l`?

**Comment by yosiat** [comment]
@nathanielc I have no idea as well.. I will try to check why TopSelectorTests are not passing and will fix the TestStream_InfluxQL tests to contain the tags in their expectations tomorrow.

**Comment by yosiat** [comment]
@nathanielc Does those are simple selectors as well?
* sum
* count
* distinct
* mean
* median

**Comment by yosiat** [comment]
@nathanielc are we good to go? If so I will squash

File: influxql.gen.go [hunk 3]
```diff
@@ -81,6 +91,10 @@ func (a *floatPointBulkAggregator) AggregateBatch(b *models.Batch) {
 			// We need to populate the Aux fields
 			floatPopulateAuxFieldsAndTags(&slice[i], a.topBottomInfo.FieldsAndTags, p.Fields, p.Tags)
 		}
+
+		if a.isSimpleSelector {
+			(&slice[i]).Aux = []interface{}{p.Tags}
```
Comment (nathanielc) [inline influxql.gen.go]: There shouldn't be any need here to take the reference of the slice element.

File: influxql.gen.go [hunk 4]
```diff
@@ -225,6 +260,10 @@ func (a *integerPointBulkAggregator) AggregateBatch(b *models.Batch) {
 			// We need to populate the Aux fields
 			integerPopulateAuxFieldsAndTags(&slice[i], a.topBottomInfo.FieldsAndTags, p.Fields, p.Tags)
 		}
+
+		if a.isSimpleSelector {
+			(&slice[i]).Aux = []interface{}{p.Tags}
```
Comment (nathanielc) [inline influxql.gen.go]: Same here

File: influxql.gen.go.tmpl [hunk 5]
```diff
@@ -42,6 +43,10 @@ func (a *{{.name}}PointAggregator) AggregateBatch(b *models.Batch) {
 			// We need to populate the Aux fields
 			{{.name}}PopulateAuxFieldsAndTags(ap, a.topBottomInfo.FieldsAndTags, p.Fields, p.Tags)
 		}
+
+    if a.isSimpleSelector {
```
Comment (nathanielc) [inline influxql.gen.go.tmpl]: Can you convert the spaces to tabs in these tmpl files since they are supposed to read as go files? Thanks

**Comment by nathanielc** [comment]
@yosiat Just those minor changes and a squash and this should be good to go. Thanks!

**Comment by nathanielc** [comment]
@yosiat  Also can you add a CHANGELOG entry? And mark it as BREAKING? Since is possible that someone was relying on the previous incorrect behavior.

**Comment by yosiat** [comment]
@nathanielc Done, can you make sure the changelog/indentation formatting is ok?

File: pipeline/influxql.gen.go.tmpl [hunk 6]
```diff
@@ -12,6 +12,7 @@ type ReduceCreater struct {
 {{end}}{{end}}{{end}}
 
 	TopBottomCallInfo *TopBottomCallInfo
+  IsSimpleSelector  bool
```
Comment (nathanielc) [inline pipeline/influxql.gen.go.tmpl]: Can you fix these spaces too?

Comment (yosiat) [inline pipeline/influxql.gen.go.tmpl]: done

**Comment by yosiat** [comment]
@nathanielc the tests are passing and the formatting is ok :+1:

**Pull request merged**